modules/database/src/std/rec/seqRecord.c

```diff
@@ -282,10 +282,10 @@ static void processCallback(epicsCallback *arg)
 static long get_units(DBADDR *paddr, char *units)
 {
     seqRecord *prec = (seqRecord *) paddr->precord;
-    int fieldOffset = dbGetFieldIndex(paddr) - indexof(DLY1);
+    int fieldOffset = dbGetFieldIndex(paddr) - indexof(DLY0);
 
     if (fieldOffset >= 0)
-        switch (fieldOffset & 2) {
+        switch (fieldOffset & 3) {
```

Comment:
I am having trouble understanding what the intent of this seqRecord meta-data mapping was and is.

- DLYx <- static `"s"`
- DOLx <- empty string?
- DOx <- dbGetUnits(DOLx)
- LNKx <- dbGetUnits(DOLx)

Reply:
The function is never called for LNKx.

Reply:
The original code ( & 2) depended on the assumption that the function is only called for fields DLYx (fieldOffset & 3 == 0) or DOx (fieldOffset & 3 == 2) but never for the links DOLx or LNKx. Thus, I had tested only & 2. But I found it "safer" to actually test both bits to ensure that the links DOLx and LNKx never return a constant value or call dbGet*() but instead don't return any units, precision, etc. The change is probably not necessary, but after the introduction of DLY0/DO0/DOL0/LNK0 broke something, I wanted to be more robust against future changes.

Reply:
So the meta-data mapping is:
 * DLYx  <- units=`"s"`; prec=seqDLYprecision
 * DOLx <- default: don't set units (i.e. whatever the caller prepared, usually `""`), prec=PREC (as before)
 * DOx  <- dbGetUnits(DOLx), dbGetPrecision(DOLx)
 * LNKx  <- default: don't set units, prec = PREC

Reply:
> The function is never called for the links LNKx or DOx. (I think)

I add `printf("%s.%s calling %s\n", paddr->precord->name, paddr->pfldDes->name, __func__);` to each of the `get_*` metadata functions in `seqRecord.c`, load a single seqRecord named `seq`, and then run:

```
caget -d GR_DOUBLE seq.{DLY0,DOL0,DO0,LNK0}
```

which causes `softIoc` to print:

```
seq.DLY0 calling get_units
seq.DLY0 calling get_precision
seq.DLY0 calling get_graphic_double
seq.DLY0 calling get_alarm_double
seq.DOL0 calling get_units
seq.DOL0 calling get_graphic_double
seq.DOL0 calling get_alarm_double
seq.DO0 calling get_units
seq.DO0 calling get_precision
seq.DO0 calling get_graphic_double
seq.DO0 calling get_alarm_double
seq.LNK0 calling get_units
seq.LNK0 calling get_graphic_double
seq.LNK0 calling get_alarm_double
```

Reply:
> So the meta-data mapping is: ...

This seems reasonable for `get_units()`, and indeed the others as well.  So I think your change is fixing a bug introduced by [27e80be2e54765000471144b7ff7067f565dd16e](https://github.com/epics-base/epics-base/commit/27e80be2e54765000471144b7ff7067f565dd16e#diff-1fb0f72ef42fe5df5cf554d9651244eeed42c1ea892245053a965cad9302e0c4R432).